### PR TITLE
Molly chat improvements

### PR DESCRIPTION
**Pull request by MegumiKim**
Molly frontend tasks I am working on can be divided to several stages.
I would like to request reviews at each sub tasks I am working on.

First commits encompass the following changes:
 - Add Label to textarea
 - Prevent empty query to be sent
 - Enable press enter query submit
 - Unhide textarea for mobile view

One question: 
Now that query can be submitted on press Enter key, can I remove submit button in order to secure more space for textarea and avoid confusion?

### CODE REVIEW

**Comment by StianSto** [comment]
great work as always :) tested it on my phone, and while molly wasnt visible at first (fixable by adjusting height on the wrapper) i was able to write and send a query, and got a response. im also not able to send an empty query. 

requests: 
- [ ] could you add the new changes to +server.ts in molly package?

**Comment by MegumiKim** [comment]
@StianSto 
Thanks Stian for testing.
As requested, uncommented +server.ts file.

File: packages/molly/src/routes/+server.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-// import MollyAI from "$lib/server/openai";
+import { OPENAI_TOKEN_LIMIT, OPENAI_API_KEY } from '$env/static/private';
+import MollyAI from '$lib/server/openai';
 
-// const molly = new MollyAI(OPENAI_API_KEY, Number(OPENAI_TOKEN_LIMIT));
+const molly = new MollyAI(OPENAI_API_KEY, Number(OPENAI_TOKEN_LIMIT));
 
-// export const GET = molly.handlers.chatCompletion.GET;
\ No newline at end of file
+export const GET = molly.handlers.chatCompletion.GET;
```
Comment (StianSto) [inline packages/molly/src/routes/+server.ts:6]:
```suggestion
export const POST = molly.handlers.chatCompletion.POST;
```

**Review by StianSto — CHANGES_REQUESTED** [review]
new backend uses POST. other than that i can't see a reason this shouldn't be good to go :)

File: packages/molly/src/routes/+server.ts (hunk 1 above)
Comment (MegumiKim) [reply packages/molly/src/routes/+server.ts]: Sorry for my oversight! Thanks for the suggestion :-)

**Comment by MegumiKim** [comment]
Molly chat box should now automatically scrolls to the bottom.

**Comment by MegumiKim** [comment]
- Molly's chat box is now automatically scrolled to the bottom.
- I have removed the submit button to see if it is better. Please let me know if you encounter any issue.

File: packages/molly/src/lib/components/MollyForm.svelte [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
-	</button>
+	<form >
+	<label for="userText">Ask Molly</label>
+		<div class="flex input-group sm:input-group-divider sm:grid-cols-[1fr_auto] rounded">
+			<textarea bind:value={query}
+			class="textarea bg-slate-100 dark:bg-slate-200 rounded-none p-1 text-black"
+			placeholder="Your Query"
+			id="userText"
+			on:keypress={(event) => handleKeyPress(event)} />
+			<!-- <button type="submit" class="btn variant-filled-primary rounded-none">
```
Comment (Fermain) [inline packages/molly/src/lib/components/MollyForm.svelte:38]: Can you remove the dead code?

File: packages/molly/src/lib/components/MollyHeader.svelte [hunk 3]
```diff
@@ -4,11 +4,8 @@
 	import closeButton from './icons/angle-small-down.svg'
 	let img: string = closeButton;
 
-	export let isOpen = false;
+export let toggleMollyOpen:()=>void;
```
Comment (Fermain) [inline packages/molly/src/lib/components/MollyHeader.svelte:7]: It would be awesome if this focused the text box

File: packages/molly/src/lib/components/MollyMessages.svelte [hunk 4]
```diff
@@ -27,4 +29,5 @@
 			}}
 		/>
 	{/if}
-</div>
\ No newline at end of file
+</div>
```
Comment (Fermain) [inline packages/molly/src/lib/components/MollyMessages.svelte:32]: Formatting/indentation?

**Review by Fermain — CHANGES_REQUESTED** [review]
Excellent work! Just a few minor changes and one new request

File: packages/molly/src/lib/components/MollyMessages.svelte (hunk 4 above)
Comment (MegumiKim) [reply packages/molly/src/lib/components/MollyMessages.svelte]: Sorry I forgot to run format.

**Comment by MegumiKim** [comment]
Please note the following changes:

- Re-organized Molly components (now Molly chat box does not live inside Molly button).
- Added custom resize to Molly chat box.
- Replace the close button to Molly icon to simplify the design. Since same icon opens the chat box, I believe this is intuitive to user...
- Added hover state (drop shadow) to Molly icon buttons. Can be less subtle if it is required. 
- Added aria-label to Molly icon btns

**Comment by Anclagen** [comment]
There are some issues when it comes to resizing the box.

On Firefox Molly is expanding with the text.
molly-firefox

On chrome its fine till you resize it.
molly-chrrome

**Comment by Fermain** [comment]
We should bind escape to closing Molly. when fully expanded, it sits behind the header meaning you can't close it.

Would be useful to have a small explainer like:

> Enter to send. Shift + Enter for new line.

**Comment by MegumiKim** [comment]
@Anclagen Thanks for spotting. 
There was a bug in the last commit. Should be ok now...

**Comment by MegumiKim** [comment]
In addition to custom resize, I added fixed resize button with placeholder icon. 
Please test if it works ok/good size.

File: packages/molly/src/lib/components/Molly.svelte
Comment (pretzL) [inline packages/molly/src/lib/components/Molly.svelte:1]: I wanted to add this as a suggestion because it might be a little hard to follow the explanation, but Git didn't want to let me do that easily so I'll try to explain. Feel free to reach out if itæs unclear.

Instead of getting the container element like so
```js
const container = document.querySelector<HTMLElement>('.chat-container')
```
You can use Svelte's own `bind` directive to reference the element, like so
```svelte
<div bind:this={container} class="chat-container etc...
```

You'd then define the `container` variable simply in the top of the script tag like so:
```svelte
let container:<HTMLElement> = null;
```

File: packages/molly/src/lib/components/MollyButton.svelte [hunk 5]
```diff
@@ -1,22 +1,13 @@
-<script>
-	import MollyHeader from './MollyHeader.svelte';
+<script lang='ts'>
+	// import MollyHeader from './MollyHeader.svelte';
```
Comment (pretzL) [inline packages/molly/src/lib/components/MollyButton.svelte:2]: Don't leave commented out code in the PR.
```suggestion
```

**Review by pretzL — CHANGES_REQUESTED** [review]
There are some formatting issues throughout the PR, which should be resolved before merging. See also my comments.

File: packages/molly/src/lib/components/Molly.svelte
Comment (Fermain) [reply packages/molly/src/lib/components/Molly.svelte:1]: This guy sveltes

Comment (pretzL) [reply packages/molly/src/lib/components/Molly.svelte:1]: 😎

**Comment by MegumiKim** [comment]
@StianSto @pretzL @Fermain 
All the requested changes are now made. Hope you could approve :-)

**Review by Fermain — APPROVED** [review]
I did some fiddling around with formatting rules, but it's now fixed and in line with the LMS formatting rules.
